tasks/prepare-binaries.js

```diff
@@ -104,36 +114,60 @@ async function main (platforms) {
     fsExtra.ensureDirSync(chromiumDir)
   }
 
-  // get browser
-  await getBrowsers(platforms)
-
-  // using pkg create the binary for asciidoctor-web-pdf
-  await createPackage(platforms)
-
+  if (isDryrun()) {
+    console.log('skipping browser download ..')
+  } else {
+    // get browser
+    await getBrowsers(platforms)
+  }
+  if (isDryrun()) {
+    console.log('skipping package creation ..')
+  } else {
+    // using pkg create the binary for asciidoctor-web-pdf
+    await createPackage(platforms)
+  }
   // copy MathJax and the css/examples/fonts folders
   copyAssets(platforms)
 
-  await archive(platforms)
+  return archive(platforms)
+}
+
+function isDryrun () {
+  return process.env.DRY_RUN !== undefined
 }
 
-// allow invoking `node tasks/prepare-binaries.js` with linux/mac/win as the argument
-// e.g. `npm run build linux`
-;(async () => {
-  try {
-    if (process.argv.length > 2) {
-      const platform = process.argv[2]
-      if (platform in platforms) {
-        const singleBuild = {}
-        singleBuild[platform] = platforms[platform]
-        await main(singleBuild)
+function exit (exitcode) {
+  if (isDryrun() && exitcode === 0) {
+    console.warn('dry run: generated output *not* useful for production.')
+  }
+  process.exit(exitcode)
+}
+
+function main () {
```

Comment:
I prefer `;(async () => { })()`, could you please revert this change?
Once we drop support for Node < 14, we will be able to use top-level await and simplify the code even more: https://v8.dev/features/top-level-await